docs/Integrating Your HMI/Vehicle Data/index.md

```diff
@@ -0,0 +1,169 @@
+                },
+                {
+                    "name": "customDeprecatedVal",
+                    "key": "KEY_CUSTOM_DEPRECATED_VAL",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true,
+                    "until": "7.0"
+                },
+                {
+                    "name": "customDeprecatedVal",
+                    "key": "KEY_CUSTOM_DEPRECATED_VAL",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true,
+                    "deprecated": true
+                    "since": "7.0"
+                }
+            ],
+            "key": "KEY_CUSTOM_STRUCT",
+            "type": "Struct",
+            "mandatory": false
+        }
+    ]
+}
+```
+
+In addition to custom items, this feature can be used to expose other vehicle data items that were introduced to the project in later versions. This can be useful when the software version on the head unit cannot be updated easily. If a vehicle data item is added into the project, the definition of this item will be included in the policy table by default. Any vehicle data items which are defined in Core's local Mobile API will be ignored from the policy table, but newer items will be interpreted as custom items. This allows apps to use these data items normally if they are exposed by the head unit, even when they were not initially supported.
+
+### Custom Vehicle Data Requests
+
+Custom vehicle data requests have a separate structure to normal vehicle data requests. While normal vehicle data items are requested using the key structure of `"<item.name>: true"`, custom items are constructed using the `item.key` field and can have a nested structure (when requesting `Struct` items). For example, when requesting all of the vehicle data items which are defined above, the HMI would receive the following message:
+
+```
+{
+    "id" : 139,
+    "jsonrpc" : "2.0",
+    "method" : "VehicleInfo.GetVehicleData",
+    "params" : {
+        "KEY_CUSTOM_STRING": true,
+        "KEY_CUSTOM_INT": true,
+        "KEY_CUSTOM_FLOAT": true,
+        "KEY_CUSTOM_BOOL": true,
+        "KEY_CUSTOM_ARRAY": true,
+        "KEY_CUSTOM_STRUCT": {
+            "KEY_CUSTOM_STRUCT_VAL": true,
+            "KEY_CUSTOM_STRUCT_VAL2": true,
+            "KEY_CUSTOM_DEPRECATED_VAL": true
+        }
+    }
+}
+```
+
+### Vehicle Data Mapping File
+
+Since these keys may not be immediately known by the HMI, a vehicle data mapping file can be used to connect these keys to actual readable values from the vehicle. The location where this file is hosted can be specified in the policy table in the `module_config.endpoints.custom_vehicle_data_mapping` field (see [Policy Endpoints](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/policy-fields/#endpoints)). The format of this file is OEM-defined.
+
```

Comment:
I think its worth documenting that this feature can be used by the phone to read raw can data https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0173-Read-Generic-Network-Signal-data.md#also-this-approach-can-also-be-used-to-read-entire-8-bytes-can-message-itself

Reply:
Seems a bit redundant IMO, since more complex data can already be represented. If you think it might be a common practice, I suppose I could just copy the example directly from the proposal.